### ISSUE TITLE
Do not call failure to destroy a mounted FS an INTERNAL_ERROR

### ISSUE THREAD

**Issue by tasleson** [stratis-storage/stratisd#1160]
What we get when we try to destroy a mounted FS

```bash
# bin/stratis filesystem destroy testing fs1
Execution failure caused by:
INTERNAL ERROR: low-level ioctl error
```
I think we should be a little more helpful to the user.

**Comment by tasleson** [comment]
Kudos on keeping the state consistent now :smiley:

**Comment by mulkieran** [comment]
Related: #1141. There are two problems here, both very deep. The first is the problem that engine errors are often mapped to D-Bus INTERNAL error when they are not actually internal errors. That whole system needs a complete overhaul. The second is that stratis just converts the low-level DM errors to Stratis errors and returns them. That needs chaining to handle properly. There are very many of these examples. I suggest filing a super-issue and putting all these related problems in a single issue (because they cannot be fixed separately).

**Comment by agrover** [comment]
That's not going to happen for 1.0, but we need to do something for the present. Right now we're getting a EngineError(DmError(NixError(EBUSY))) or something like that, resulting in the INTERNAL ERROR, but we should be able to catch and convert this to a StratisError with a useful "device is busy" error message, that is a Dbus::ERROR, not a Dbus::INTERNAL_ERROR.

**Comment by agrover** [comment]
Use of error-chain in dm-rs causes some complications to what I was thinking in previous comment. Getting through the IoctlError to the underlying chained io:Error is unpleasant and I can't get it to compile :frowning_face:. Instead I propose https://github.com/stratis-storage/devicemapper-rs/pull/373, and circle back for a permanent solution after 1.0.

We also may want to not use `DbusErrorEnum::INTERNAL_ERROR` in the case of this... well maybe in *all* cases. What is it other than a scarier version of ERROR? (actually, for the current issue, `DbusErrorEnum::BUSY` would be ideal but I don't see how to easily have that happen.)

**Comment by mulkieran** [comment]
Do something for the present to solve what problem?

**Comment by agrover** [comment]
> Do something for the present to solve what problem?

The non-helpful error message.

**Comment by mulkieran** [comment]
The thing that will fix our problem seems to have been merged into Rust nightly 11 days ago: https://doc.rust-lang.org/nightly/std/error/trait.Error.html.

**Comment by mulkieran** [comment]
There is more than one non-helpful message...I would stake my life on it.

**Comment by mulkieran** [comment]
So, at what point will something that was merged into nightly 11 days ago start to do us any good?

**Comment by agrover** [comment]
Included in a Rust stable release around March 2019, and then however long for it to get packaged for distros we care about.

**Comment by mulkieran** [comment]
This whole situation is most dispiriting.

**Comment by agrover** [comment]
will be fixed by #1179 and #1167

**Comment by mulkieran** [comment]
Seems like I'm working on this.

**Comment by agrover** [comment]
fixed by #1167
